pkg/resourcemanager/azuresql/azuresqlserver/azuresqlserver_reconcile.go

```diff
@@ -122,6 +122,13 @@ func (s *AzureSqlServerManager) Ensure(ctx context.Context, obj runtime.Object,
 	if instance.Status.Provisioning ||
 		(!specHashWasEmpty && instance.Status.SpecHash == hash) {
 
+		// TODO: This is a bit of a hack because really this check should be in the SDK.
+		// TODO: See: https://github.com/Azure/azure-sdk-for-go/issues/10712
+		if instance.Spec.ResourceGroup == "" {
+			instance.Status.Message = "A non-empty resource group must be specified."
```

Comment:
Why do we need this check if we have the validation now? The validation should catch the case if the RG was missing or empty, correct?

Reply:
Yeah that's a good question.

I've added a comment to the code which explains the situation, but basically the only way that this can happen is if an entity in etcd already exists without the `ResourceGroup` field set. This can happen due to an upgrade from an older version of ASO, or if the customer has disabled our conversion webhooks and put say an `v1alpha1` version of this resource into etcd rather than a `v1beta1` which is what we are using as the storage version.

This is possible because we renamed from `resourcegroup` to `resourceGroup` in the YAML/JSON for this resource. 

This check is really probably unlikely to be hit in most cases because it's kinda hard to be in a situation that causes it (but it is possible see #1106 which was caused (I think) due to conversion webhooks being disabled and using a non-default version which caused this to be empty - basically the field was in the CRD (so validation of CRD passes), but the field is not capitalized correctly (due to difference in CRD spec between v1alpha1 and v1beta1), so when we go to actually use it here we get empty string.

Reply:
After some more testing we actually don't even need this check, because missing RG is caught even in the case of older API versions earlier in the stack. I've removed this